huaweicloud/services/acceptance/modelarts/resource_huaweicloud_modelarts_authorization_test.go

```diff
@@ -0,0 +1,158 @@
+package modelarts
+
+import (
+	"encoding/json"
+	"fmt"
+	"strings"
+	"testing"
+
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/resource"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/terraform"
+
+	"github.com/chnsz/golangsdk"
+	"github.com/chnsz/golangsdk/pagination"
+
+	"github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/config"
+	"github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/services/acceptance"
+	"github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/services/modelarts"
+)
+
+func getModelArtsAuthorizationResourceFunc(cfg *config.Config, state *terraform.ResourceState) (interface{}, error) {
+	region := acceptance.HW_REGION_NAME
+	// getModelAuth: Query the ModelArts authorization.
+	var (
+		getModelAuthHttpUrl = "v2/{project_id}/authorizations"
+		getModelAuthProduct = "modelarts"
+	)
+	getModelAuthClient, err := cfg.NewServiceClient(getModelAuthProduct, region)
+	if err != nil {
+		return nil, fmt.Errorf("error creating ModelArts Client: %s", err)
+	}
+
+	getModelAuthPath := getModelAuthClient.Endpoint + getModelAuthHttpUrl
+	getModelAuthPath = strings.ReplaceAll(getModelAuthPath, "{project_id}", getModelAuthClient.ProjectID)
+
+	getModelAuthResp, err := pagination.ListAllItems(
+		getModelAuthClient,
+		"offset",
+		getModelAuthPath,
+		&pagination.QueryOpts{MarkerField: ""})
+	if err != nil {
+		return nil, fmt.Errorf("error retrieving ModelArts authorization: %s", err)
+	}
+
+	getModelAuthRespJson, err := json.Marshal(getModelAuthResp)
+	if err != nil {
+		return nil, fmt.Errorf("error retrieving ModelArts authorization: %s", err)
```

Comment:
```suggestion
		return nil, fmt.Errorf("error marshal ModelArts authorization: %s", err)
```

Reply:
no, it's an internal error, it is difficult for the user to understand, just need to explain at what stage the error occurred